packages/common/src/api/OmSupplyApiContext.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { FC, useEffect, useState, useCallback } from 'react';
+import { createContext } from 'react';
+import { GraphQLClient } from 'graphql-request';
+import { getSdk } from '../types';
+
+export const createOmSupplyApi = (
+  url: string
+): { api: OmSupplyApi; client: GraphQLClient } => {
+  const client = new GraphQLClient(url);
+  const api = getSdk(client);
+  return { client, api };
+};
+
+export type OmSupplyApi = ReturnType<typeof getSdk>;
+
+interface OmSupplyApiControl {
+  api: OmSupplyApi;
+  client: GraphQLClient;
+  setHeader: (header: string, value: string) => void;
+  setUrl: (url: string) => void;
+}
+
+const OmSupplyApiContext = createContext<OmSupplyApiControl>({
+  ...createOmSupplyApi(''),
```

Comment:
Hopefully setting this with an empty url initially is fine. I didn't want optionals in the types.. 😁 
 We pass the correct url from `Environment` straight away however, I can see that it might be better to just use that URL here rather than passing it as a prop. 

Though I suspect this to not be a problem, I'm not sure how we're going to 'get' that URL in production and whether using the hardcoded string would work 🤔

Reply:
yes, getting the URL in production will be interesting and I'm not sure yet how that might look.. good not to assume that we can source it immediately from `Environment` or load it from a text file